odoo/addons/base/models/res_users.py

```diff
@@ -1618,6 +1619,26 @@ def remove(self):
             return {'type': 'ir.actions.act_window_close'}
         raise AccessError(_("You can not remove API keys unless they're yours or you are a system user"))
 
+    def _check_credentials(self, key):
+        index = key[:INDEX_SIZE]
+        self.env.cr.execute('SELECT id, key FROM res_users_apikeys WHERE index = %s', [index])
+        for key_id, current_key in self.env.cr.fetchall():
+            if KEY_CRYPT_CONTEXT.verify(key, current_key):
+                return self.browse(key_id)
+
+    def _generate(self, name, scope=None):
```

Comment:
Thinking about it more, because these parameters are both optional strings, the callsite is pretty muddled. They should be keyword-only and required. I'd also put `scope` first as it's the required-est of the two:

```suggestion
    def _generate(self, *, scope, name):
```
with a docstring doing basic documentation and noting that both *can* be null, with the implication this has (a null scope meaning the key will be accepted by every endpoint).